### PR TITLE
mrc-user-exists fix user exists issue with oauth method

### PR DESCRIPTION
**Pull request by LekanAnni**
## Description

This PR fixes issue with checking if an oauth2 user exists. Currently checking if a user exists for both db and oauth users using profileService which throws user does not exists for oauth users, error thrown because oauth is not required to use profile service interface to interact with database users. PR now supports checking user existence for both db and oauth users, respectively.

## Type of version change
_Delete as appropriate_

Major - Minor - Patches - None

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

**Review by r-ash — CHANGES_REQUESTED** [review]
Deployed this to dev and it finds the user but then fails to share the project with them

image

Is there a test which checks sharing a project with users without a serializedprofile?

**Comment by LekanAnni** [comment]
> Deployed this to dev and it finds the user but then fails to share the project with them
> 
> image
> 
> Is there a test which checks sharing a project with users without a serializedprofile?

I left usage of userLogic getUser() in projectController outdated. This has now been updated and test added.

Once browser tests get in, we can run browser tests with new users. However, I have created new users and shared projects with them, works as expected.

**Review by EmmaLRussell — COMMENTED** [review]
I haven't run this up, as it seems it's getting well manually tested anyway by @r-ash. Code makes sense to me, couple of tiny comments - neither of which are showstoppers!

File: src/app/src/main/kotlin/org/imperial/mrc/hint/logic/UserLogic.kt [hunk 1]
```diff
@@ -62,7 +63,7 @@ class DbProfileServiceUserLogic(private val userRepository: UserRepository,
         profileService.removeById(user.id)
     }
 
-    override fun getUser(email: String): CommonProfile?
+    override fun getUser(email: String, oauthLogin: Boolean): CommonProfile?
```
[the remark this answers is not in the document]
Comment (LekanAnni) [reply src/app/src/main/kotlin/org/imperial/mrc/hint/logic/UserLogic.kt:66]: OAuth20 profile is an implementation of CommonProfile but yeah we don't need commonProfile as it's only a true of false checking. The profile is still needed for Guest login and when login method is DB.

[the remark this answers is not in the document]
Comment (LekanAnni) [reply src/app/src/main/kotlin/org/imperial/mrc/hint/logic/UserLogic.kt:66]: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-3878/Change-return-type-for-getUser-in-UserLogic-to-Boolean

**Review by EmmaLRussell — APPROVED** [review]
Code looks good to me, pending @r-ash 's full testing on server

**Comment by LekanAnni** [comment]
We confirmed this is working on server.

**Pull request merged**